eppo_client/client.py

```diff
@@ -31,9 +34,47 @@ def __init__(
         )
         self.__poller.start()
 
+    def get_string_assignment(
+        self, subject_key: str, flag_key: str, subject_attributes=dict()
+    ) -> Optional[str]:
+        assigned_variation = self.get_assignment_variation(subject_key, flag_key, subject_attributes, VariationType.STRING)
+        return assigned_variation.typedValue if assigned_variation is not None else assigned_variation
+
+    def get_numeric_assignment(
+        self, subject_key: str, flag_key: str, subject_attributes=dict()
+    ) -> Optional[Number]:
+        assigned_variation = self.get_assignment_variation(subject_key, flag_key, subject_attributes, VariationType.NUMERIC)
+        return assigned_variation.typedValue if assigned_variation is not None else assigned_variation
+
+    def get_boolean_assignment(
+        self, subject_key: str, flag_key: str, subject_attributes=dict()
+    ) -> Optional[bool]:
+        assigned_variation = self.get_assignment_variation(subject_key, flag_key, subject_attributes, VariationType.BOOLEAN)
+        return assigned_variation.typedValue if assigned_variation is not None else assigned_variation
+
+    def get_parsed_json_assignment(
+        self, subject_key: str, flag_key: str, subject_attributes=dict()
+    ) -> Optional[Dict[Any, Any]]:
+        assigned_variation = self.get_assignment_variation(subject_key, flag_key, subject_attributes, VariationType.JSON)
+        return assigned_variation.typedValue if assigned_variation is not None else assigned_variation
+
+    def get_json_string_assignment(
+        self, subject_key: str, flag_key: str, subject_attributes=dict()
+    ) -> Optional[str]:
+        assigned_variation = self.get_assignment_variation(subject_key, flag_key, subject_attributes, VariationType.JSON)
+        return assigned_variation.value if assigned_variation is not None else assigned_variation
+
+    # deprecated in favor of the typed get_<type>_assignment methods
```

Comment:
Is this comment sufficient as a deprecation warning? I was also considering this https://pypi.org/project/deprecation/

Reply:
Oh, this is better: `typing` has a `@deprecated` decorator! https://peps.python.org/pep-0702/#example

Will do this

Reply:
I think this comment itself is fine